numba/dppl/withcontexts.py

```diff
@@ -0,0 +1,144 @@
+from numba.core import compiler, typing, types, sigutils
+from numba.core.compiler_lock import global_compiler_lock
+from numba.core.dispatcher import _DispatcherBase
+from numba.core.transforms import find_region_inout_vars
+from numba.core.withcontexts import (WithContext, _mutate_with_block_callee, _mutate_with_block_caller,
+                                     _clear_blocks)
+from numba.dppl.compiler import DPPLCompiler
+from numba.core.cpu_options import ParallelOptions
+
+
+class _DPPLContextType(WithContext):

```

Comment:
Can you please add some comments to explain what this functions are doing?

Reply:
This is a class that creates a separate dispatcher for the lifted code with our semantics. This is needed to run the code with the _'offload': True_ option and our pipeline